examples/sdk-upload/README.md

```diff
@@ -0,0 +1,334 @@
+[above](#step-1---creating-the-bucket). Take a look at [the tutorial](https://github.com/dwyl/image-uploads/blob/master/examples/direct-upload/README.md#step-2---creating-an-iam-user-with-s3-permissions)
+and again follow the steps.
+
+### Step 3 - Configure the AWS SDK
+
+In order to upload to our S3 bucket, we have to use the AWS SDK. Install it using
+the following command:
+
+`$ npm install aws-sdk --save`
+
+Next we'll need to configure our AWS SDK. You can do this in a number of ways.
+We're going to create a credentials file at `~/.aws/credentials`. To do this we
+take the following steps:
+
+* `$ cd` this takes us back to the root of our file system
+* `$ mkdir .aws` creates a folder that will hold our credentials file
+* `$ cd mkdir` navigate to our `.aws` directory
+* `$ touch credentials` creates our credentials file
+* Open your credentials file in your text editor and add the following:
+```
+[default]
+aws_access_key_id = your_access_key_id
+aws_secret_access_key = your_secret_key
+```
+* save the file
+
+The credentials should be the ones associated with the IAM user you created in
+[Step 2]((#step-2---creating-an-iam-user-with-s3-permissions)).
+
+#### You will now be able to use the AWS SDK!
+
+### Step 4 - Implement the SDK
+
+First you'll need to set some environment variables. These are as follows:
+
+`export S3_REGION=<YOUR_REGION>`
+`export S3_BUCKET=<YOUR_BUCKET_NAME>`
+
+Create a `src` directory in the root of your project. Create a file within this
+directory called `upload.js`. This file will contain our SDK functionality.
+Add the following to this file:
+
+```js
+// load the AWS SDK
+var AWS = require('aws-sdk')
+// crypto is used for our unique filenames
+var crypto = require('crypto')
+// we use path to add the relative file extension
+var path = require('path')
+
+// assign our region from our environment variables
+AWS.config.region = process.env.S3_REGION
+
+/**
+* Returns data specific to our upload that we send to the front end
+* @param {Buffer} file - file that we are uploading
+* @param {string} filename - name of the file to be uploaded
+* @param {function} callback
+**/
+function upload (file, filename, callback) {
+  // creating our new filename
+  var filenameHex =
+  filename.split('.')[0] +
+  crypto.randomBytes(8).toString('hex') +
+  path.extname(filename)
+  // loading our bucket name from our environment variables
+  var bucket = process.env.S3_BUCKET
+  // creating a new instance of our bucket
+  var s3Bucket = new AWS.S3({params: {Bucket: bucket}})
+  // sets the params needed for the upload
+  var params = {Bucket: bucket, Key: filenameHex, Body: file}
+  // SDK upload to S3
+  s3Bucket.upload(params, function (err, data) {
+    if (err) console.log(err, data)
+    // callback with the data that gets returned from S3
+    else callback(null, data)
+  })
+}
+
+module.exports = {
+  upload
+}
+```
+
+#### We've now set up our S3 upload function!
```

Comment:
I feel as though it helps to separate the sections. I can remove it if you'd like